### PR TITLE
feat(BOS-285): drop catchup subscriptions on unexpected error

### PR DESCRIPTION
**Pull request by dariusandz**
This change is [Reviewable](https://reviewable.io/reviews/inventi/eventstore-tools/44)

### CODE REVIEW

File: src/test/kotlin/io/inventi/eventstore/eventhandler/util/WithMeterRegistryAssertions.kt [hunk 1]
```diff
@@ -0,0 +1,29 @@
...
+
+interface WithMeterRegistryAssertions {
+    var meterRegistry: MeterRegistry
+
+    private val assertDelta: Double
+        get() = 1e-6
+
+    fun assertGaugeWithTagHasValue(gaugeName: String, meterTag: String, meterValue: Double) {
+        Awaitility.await().pollDelay(1, TimeUnit.SECONDS).atMost(5, TimeUnit.SECONDS).untilAsserted {
+            assertEquals(valueOfGaugeWithTag(gaugeName, meterTag), meterValue, assertDelta)
```
Comment (Zilv1nas) [inline src/test/kotlin/io/inventi/eventstore/eventhandler/util/WithMeterRegistryAssertions.kt:17]: I suggest to use the same assertion library as in other tests: `valueOfGaugeWithTag(gaugeName, meterTag) shouldBeEqualTo meterValue`

File: src/test/kotlin/io/inventi/eventstore/eventhandler/util/WithMeterRegistryAssertions.kt [hunk 2]
```diff
@@ -0,0 +1,29 @@
...
+        }
+    }
+
+    private fun valueOfGaugeWithTag(gaugeName: String, meterTag: String): Double {
+        return meterRegistry.find(gaugeName)
```
Comment (Zilv1nas) [inline src/test/kotlin/io/inventi/eventstore/eventhandler/util/WithMeterRegistryAssertions.kt:22]: Meter registry has api to look for tags:
```
return meterRegistry.find(gaugeName)
    .tag(tagName, tagValue)
    .gauge()!!
    .value()
```

File: src/test/kotlin/io/inventi/eventstore/eventhandler/CatchupSubscriptionsMeterRegistryIntegrationTest.kt [hunk 3]
```diff
@@ -0,0 +1,94 @@
...
+    override lateinit var eventStore: EventStore
+
+    @Autowired
+    override lateinit var meterRegistry: MeterRegistry
+
+    @Autowired
+    private lateinit var handler: Handler
+
+    @Test
+    fun `when handler is subscribed to stream it's gauge value is seen as active`() {
```
Comment (Zilv1nas) [inline src/test/kotlin/io/inventi/eventstore/eventhandler/CatchupSubscriptionsMeterRegistryIntegrationTest.kt:44]: this naming does not follow our usual conventions, e.g. `shows metric's gauge as active when handler is subscribed to stream`

**Review by Zilv1nas — CHANGES_REQUESTED** [review]

File: src/test/kotlin/io/inventi/eventstore/eventhandler/util/WithMeterRegistryAssertions.kt (hunk 1 above)
Comment (dariusandz) [reply src/test/kotlin/io/inventi/eventstore/eventhandler/util/WithMeterRegistryAssertions.kt]: Done

File: src/test/kotlin/io/inventi/eventstore/eventhandler/util/WithMeterRegistryAssertions.kt (hunk 2 above)
Comment (dariusandz) [reply src/test/kotlin/io/inventi/eventstore/eventhandler/util/WithMeterRegistryAssertions.kt:19]: Nice, done

File: src/test/kotlin/io/inventi/eventstore/eventhandler/CatchupSubscriptionsMeterRegistryIntegrationTest.kt [hunk 4]
```diff
@@ -0,0 +1,100 @@
...
+    override lateinit var eventStore: EventStore
+
+    @Autowired
+    override lateinit var meterRegistry: MeterRegistry
+
+    @Autowired
+    private lateinit var handler: Handler
+
+    @Test
+    fun `show metric's gauge as active when handler is subscribed to stream`() {
```
Comment (Zilv1nas) [inline src/test/kotlin/io/inventi/eventstore/eventhandler/CatchupSubscriptionsMeterRegistryIntegrationTest.kt:45]: shows*

**Review by Zilv1nas — APPROVED** [review]

**Pull request merged**